packages/core/getBusyTimes.ts

```diff
@@ -201,23 +204,43 @@ export async function getBusyTimes(params: {
   performance.measure(`prisma booking get took $1'`, "prismaBookingGetStart", "prismaBookingGetEnd");
   if (credentials?.length > 0) {
     const startConnectedCalendarsGet = performance.now();
-    const calendarBusyTimes = await getBusyCalendarTimes(
+    const calendarBusyTimesWithTimeBlocks = await getBusyCalendarTimes(
       username,
       credentials,
       startTime,
       endTime,
-      selectedCalendars
+      selectedCalendars,
+      timeBlocksArr?.length ? true : false
     );
     const endConnectedCalendarsGet = performance.now();
     logger.debug(
       `Connected Calendars get took ${
         endConnectedCalendarsGet - startConnectedCalendarsGet
       } ms for user ${username}`,
       JSON.stringify({
-        calendarBusyTimes,
+        calendarBusyTimesWithTimeBlocks,
       })
     );
 
+    const calendarBusyTimes: EventBusyData[] = [];
+    if (timeBlocksArr?.length) {
+      // separate the calendar events that contain any of the time block values in the event title
```

Comment:
Remove this comment, it's understandable

Reply:
Done